pkg/models/usertoken/user_token.go

```diff
@@ -35,3 +36,21 @@ type UserToken struct {
 	RevokedAt     int64
 	UnhashedToken string
 }
+
+const UrgentRotateTime = 1 * time.Minute
+
+func (t *UserToken) NeedRotation(rotationInterval time.Duration) bool {
+	rotatedAt := time.Unix(t.RotatedAt, 0)
+	if !t.AuthTokenSeen {
+		return rotatedAt.Before(time.Now().Add(-UrgentRotateTime))
+	}
+
+	return rotatedAt.Before(time.Now().Add(-rotationInterval))
+}
+
+const rotationLeeway = 5 * time.Second
```

Comment:
Should me make it configurable ? (In case some servers have poor time sync)

Reply:
That sounds reasonable, would it be ok to do in a seperate pr?

Reply:
Sure, this PR is already huge